### ISSUE TITLE
`DisputesClient` rejecting `java.time.Instant`'s in `DisputesQueryFilter`

### ISSUE THREAD

**Issue by kdrakon** [checkout/checkout-sdk-java#166]
### Environment

* Checkout SDK version: 5.4.0

### Description

When using the disputes client `query` method, with the parameter of a `DisputesQueryFilter`, the client gets a rejection from the REST API when specifying the `from` and/or `to` fields (which are of type `java.time.Instant`). 

### Expected behavior

The client should accept the `Instant` values set for `from` and `to` and return any disputes updated within the time window.

### Current behavior

For example, assuming `now` is from `Instant.now()`, the following filter will fail:
```scala
DisputesQueryFilter.builder().from(now.minusSeconds(3600)).to(now).build()
```
The Java exception returned is like this:
```
CheckoutApiException(requestId=0HMF57UGVGS7I:00000014, httpStatusCode=422, errorDetails={request_id=0HMF57UGVGS7I:00000014, error_type=request_invalid, error_codes=[paging_from_invalid_format, paging_to_invalid_format]})
```

Debugging the parameters built from the builder, I can see the following is being sent to the REST API:
![image](https://user-images.githubusercontent.com/1347374/152902016-945d7506-4bce-4a20-b85b-9b9ce0e7ec07.png)
Notice the precision after the seconds.

If I modify `now` to be `now.truncatedTo(ChronoUnit.SECONDS)`, the parameters change to:
![image](https://user-images.githubusercontent.com/1347374/152902150-35296718-1a4d-400b-a6a7-a3b45ef62327.png)
and the request succeeds.

In conclusion, it looks like the CKO REST API does not like the additional precision of the ISO8601 timestamp and thus rejecting it as a `paging_from_invalid_format` or `paging_to_invalid_format`.

### Possible solution
It seems wrong for the client code to truncate the exact `Instant` that the implementor is creating. To me, it seems the REST API is incorrectly rejecting the timestamp because they are in fact ISO8601 compatible, albeit a little more precise.

**Comment by rsaestrela** [comment]
Hi @kdrakon. This is definitely an issue. We noticed and fixed this issue recently in the CKO .NET SDK to avoid the additional precision when serializing DateTime objects. Unfortunately, we didn't notice this issue in this SDK too.

We are going to provide and release a fix to this issue along with a few improvements on this SDK tomorrow, worst case next Thursday 10th. Thanks for reporting this!

**Comment by rsaestrela** [comment]
Hi @kdrakon. Version 5.5.0 was just released containing a fix for this issue. Just a caveat, coincidentally both Disputes modules were merged into 1 (there were different interfaces for Default and Four) - due to the similarity, code is now reused and there will be only one Dispute client interface under `com.checkout.disputes`. If you're using `com.checkout.disputes.four.*` a simple refactor to `com.checkout.disputes.*` should fix your compilation issues. Thanks for reporting this issue!

**Comment by kdrakon** [comment]
thanks @rsaestrela  for the very quick turnaround on this 🙏
